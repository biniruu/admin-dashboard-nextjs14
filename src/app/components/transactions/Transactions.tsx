import Image from 'next/image'

import styles from './transactions.module.css'

const tableHead: string[] = ['name', 'status', 'date', 'amount']

function Transactions() {
  return (
    <div className="rounded-default bg-bg-soft p-5">
      <h2 className="mb-5 font-extralight text-text-soft">transactions</h2>
      <table className={`w-full ${styles.table}`}>
        <thead>
          <tr>
            {tableHead.map(item => (
              <td key={item}>{item}</td>
            ))}
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <Image src="/noavatar.png" alt="" width={40} height={40} className={styles['user-image']} />
              john doe
            </td>
            <td>
              <span className={`${styles.status} bg-[#f7cb7375]`}>pending</span>
            </td>
            <td>14.02.2024</td>
            <td>$3.200</td>
          </tr>
          <tr>
            <td>
              <Image src="/noavatar.png" alt="" width={40} height={40} className={styles['user-image']} />
              john doe
            </td>
            <td>
              <span className={`${styles.status} bg-[#afd6ee75]`}>done</span>
            </td>
            <td>14.02.2024</td>
            <td>$3.200</td>
          </tr>
          <tr>
            <td>
              <Image src="/noavatar.png" alt="" width={40} height={40} className={styles['user-image']} />
              john doe
            </td>
            <td>
              <span className={`${styles.status} bg-[#f7737375]`}>cancelled</span>
            </td>
            <td>14.02.2024</td>
            <td>$3.200</td>
          </tr>
        </tbody>
      </table>
    </div>
  )
}

export default Transactions
