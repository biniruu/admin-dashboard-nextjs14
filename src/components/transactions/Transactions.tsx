import Image from 'next/image'

import transactionItems from './transactionItems'
import styles from './transactions.module.css'

const tableHead: string[] = ['name', 'status', 'date', 'amount']

const bgColor = {
  pending: 'bg-[#f7cb7375]',
  done: 'bg-[#afd6ee75]',
  cancelled: 'bg-[#f7737375]',
}

function Transactions() {
  return (
    <div className="rounded-default bg-bg-soft p-5">
      <h2 className="mb-5 font-extralight text-text-soft">transactions</h2>
      <table className={`w-full ${styles.table}`}>
        <thead>
          <tr>
            {tableHead.map(item => (
              <td key={item}>{item}</td>
            ))}
          </tr>
        </thead>
        <tbody>
          {/* TODO: Update this when getting data from DB is enabled */}
          {transactionItems.map(item => {
            const { img, name, status, date, price } = item
            const bg = bgColor[status as keyof typeof bgColor]

            return (
              <tr key={`${name}${status}`}>
                <td>
                  <div className="flex items-center">
                    <Image
                      src={img || '/noavatar.png'}
                      alt=""
                      width={40}
                      height={40}
                      className={`${styles['user-image']} mr-4`}
                    />
                    {name}
                  </div>
                </td>
                <td>
                  <span className={`${styles.status} ${bg}`}>{status}</span>
                </td>
                <td>{date}</td>
                <td>${price}</td>
              </tr>
            )
          })}
        </tbody>
      </table>
    </div>
  )
}

export default Transactions
