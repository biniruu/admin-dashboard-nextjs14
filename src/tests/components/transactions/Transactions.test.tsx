import renderer from 'react-test-renderer'

import Transactions from '@components/transactions/Transactions'

describe('Transactions component', () => {
  test('should render the component', () => {
    const result = renderer.create(<Transactions />).toJSON()

    expect(result).toMatchSnapshot()
  })
})
