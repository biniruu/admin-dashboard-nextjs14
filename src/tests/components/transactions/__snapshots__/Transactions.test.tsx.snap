// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions component should render the component 1`] = `
<div
  className="rounded-default bg-bg-soft p-5"
>
  <h2
    className="mb-5 font-extralight text-text-soft"
  >
    transactions
  </h2>
  <table
    className="w-full table"
  >
    <thead>
      <tr>
        <td>
          name
        </td>
        <td>
          status
        </td>
        <td>
          date
        </td>
        <td>
          amount
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            className="flex items-center"
          >
            <img
              alt=""
              className="user-image mr-4"
              data-nimg="1"
              decoding="async"
              height={40}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fnoavatar.png&w=96&q=75"
              srcSet="/_next/image?url=%2Fnoavatar.png&w=48&q=75 1x, /_next/image?url=%2Fnoavatar.png&w=96&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={40}
            />
            john doe
          </div>
        </td>
        <td>
          <span
            className="status bg-[#f7cb7375]"
          >
            pending
          </span>
        </td>
        <td>
          14.02.2024
        </td>
        <td>
          $
          3.200
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="flex items-center"
          >
            <img
              alt=""
              className="user-image mr-4"
              data-nimg="1"
              decoding="async"
              height={40}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fnoavatar.png&w=96&q=75"
              srcSet="/_next/image?url=%2Fnoavatar.png&w=48&q=75 1x, /_next/image?url=%2Fnoavatar.png&w=96&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={40}
            />
            john doe
          </div>
        </td>
        <td>
          <span
            className="status bg-[#afd6ee75]"
          >
            done
          </span>
        </td>
        <td>
          14.02.2024
        </td>
        <td>
          $
          3.200
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="flex items-center"
          >
            <img
              alt=""
              className="user-image mr-4"
              data-nimg="1"
              decoding="async"
              height={40}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fnoavatar.png&w=96&q=75"
              srcSet="/_next/image?url=%2Fnoavatar.png&w=48&q=75 1x, /_next/image?url=%2Fnoavatar.png&w=96&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={40}
            />
            john doe
          </div>
        </td>
        <td>
          <span
            className="status bg-[#f7737375]"
          >
            cancelled
          </span>
        </td>
        <td>
          14.02.2024
        </td>
        <td>
          $
          3.200
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
